lib/src/folding/parsers/fallback.dart

```diff
@@ -67,7 +80,8 @@ class FallbackFoldableBlockParser extends TextFoldableBlockParser {
     String tail = '';
 
     for (final code in text.runes) {
-      tail += String.fromCharCode(code); // ignore: use_string_buffers
+      final char = String.fromCharCode(code);
+      tail += char; // ignore: use_string_buffers
```

Comment:
Maybe we can also accumulate future characters in `head` in a similar way so we can look 'into the future' to check if a character starts a comment?

This is complexity though. We should think carefully.

Reply:
With `head` we can get rid of `tail` because they serve the same purpose.
It doesn't add a lot of complexity and allows us to get rid of a confusing flag.

I think it is better.